sdk/src/main/java/io/dapr/client/DaprClientImpl.java

```diff
@@ -1175,6 +1178,47 @@ public Mono<UnsubscribeConfigurationResponse> unsubscribeConfiguration(Unsubscri
     }
   }
 
+  @Override
+  public <T extends Message> Mono<Void> scheduleJobAlpha1(Job<T> job) {
+	  try {
+		  final String name = job.getName();
+		  final T data = job.getData();
+		  
+		  if (name == null || name.trim().isEmpty()) {
+			  throw new IllegalArgumentException("Job name cannot be null or empty");
+	  }
+	  if (data == null) {
+		  throw new IllegalArgumentException("Job data cannot be empty");
+		  }
+		  
+		  DaprProtos.Job.Builder jobBuilder = DaprProtos.Job.newBuilder()
+				  .setName(name)
+				  .setData(Any.pack(data));
```

Comment:
@artursouza here I have to convert data which is generic to Any data type, it is allowing only generics that extend Message

Reply:
@charan2628 this is only because you defined that in the DaprClient interface. Your Job class doesn't need to extend Message. Check the other domain objects such as: https://github.com/dapr/java-sdk/blob/b167c08b71c2b65cc1685ef167930ed028146dbb/sdk/src/main/java/io/dapr/client/domain/GetStateRequest.java , this is what @artursouza means by make it GRPC agnostic. You receive a GRPC message inside and translate that to a class that doesn't have any dependency to GRPC messages.